### PR TITLE
VM-Image-Preview-Check-Enhancements-4054

### PR DESCRIPTION
**Pull request by StartAutomating**

### CODE REVIEW

File: arm-ttk/testcases/deploymentTemplate/Virtual-Machines-Should-Not-Be-Preview.test.ps1 [hunk 1]
```diff
@@ -10,28 +10,15 @@ param(
...
+foreach ($sp in $storageProfiles) {
+    $storageProfile = $sp.StorageProfile
+    if (-not $storageProfile.imageReference) {
+        Write-Output "Virtual machine resource '$($sp.ParentObject.Name)' has no image to reference" -TargetObject $sp # VMSS scale up does not have a imageRef by design
     }
 
-    if ($imageReference -like '*-preview' -or $imageReference.version -like '*-preview') {
-        Write-Error "Virtual machine resource $($resource.Name) must not use a preview image" -TargetObject $ResourceType -ErrorId VM.Using.Preview.Image
+    if ($storageProfile.imageReference -like '*-preview' -or $storageProfile.imageReference.version -like '*-preview') {
+        Write-Error "Virtual machine resource '$($sp.ParentObject.Name)' must not use a preview image" -TargetObject $sp -ErrorId VM.Using.Preview.Image
```
Comment (bmoore-msft) [inline arm-ttk/testcases/deploymentTemplate/Virtual-Machines-Should-Not-Be-Preview.test.ps1:22]: "storageProfile for resource '$($sp.ParentObject.Name)' must not use a preview version" -TargetObject $sp -ErrorId VM.Using.Preview.Image

File: arm-ttk/testcases/deploymentTemplate/Virtual-Machines-Should-Not-Be-Preview.test.ps1 [hunk 2]
```diff
@@ -10,28 +10,15 @@ param(
 $TemplateObject
 )
 
-#? Should this be using Find-JsonContent?
+$storageProfiles = Find-JsonContent -Key storageProfile -InputObject $TemplateObject
```
Comment (bmoore-msft) [inline arm-ttk/testcases/deploymentTemplate/Virtual-Machines-Should-Not-Be-Preview.test.ps1:13]: let's add coverage for and a test case for

            "properties": {
                "storageProfile": "[variables('imgref')]"
            }

Where we check the variable value for preview

File: arm-ttk/testcases/deploymentTemplate/Virtual-Machines-Should-Not-Be-Preview.test.ps1 (hunk 1 above)
Comment (StartAutomating) [reply arm-ttk/testcases/deploymentTemplate/Virtual-Machines-Should-Not-Be-Preview.test.ps1]: Message updated

File: arm-ttk/testcases/deploymentTemplate/Virtual-Machines-Should-Not-Be-Preview.test.ps1 (hunk 2 above)
Comment (StartAutomating) [reply arm-ttk/testcases/deploymentTemplate/Virtual-Machines-Should-Not-Be-Preview.test.ps1:13]: Test refactored and added.

**Pull request merged**